tests/integration/test_integration.py

```diff
@@ -336,21 +336,25 @@ def test_priority_sampling_response(encoding, monkeypatch):
     s.set_tag("env", "my-env")
     s.finish()
     assert "service:my-svc,env:my-env" not in t._writer._priority_sampler._by_service_samplers
-    t.shutdown()
+    t.flush()
 
-    # For some reason the agent doesn't start returning the service information
-    # immediately
-    import time
+    if "service:my-svc,env:my-env" in t._writer._priority_sampler._by_service_samplers:
```

Comment:
what about flipping this, `if "" not in t._writer:` then we sleep and send traces, then you do the shutdown + assertion outside of the branch ?

Reply:
sure, i don't have a preference